services/QuillLMS/client/app/bundles/Shared/utils/appSettingAPIs.ts

```diff
@@ -0,0 +1,28 @@
+import { mainApiFetch, requestFailed, handleRequestErrors } from './../../Staff/helpers/comprehension';
+
+interface HandleHasAppSettingArgs {
+  appSettingSetter: (value: boolean) => void,
+  errorSetter: (value: any) => void,
+  key: string,
+}
+
+export const handleHasAppSetting = async (args: HandleHasAppSettingArgs) => {
+  const response = await mainApiFetch(`app_settings/${args.key}`);
+
+  const { status } = response;
+
+  if(requestFailed(status)) {
+    const returnedErrors = [`AppSetting HTTP status: ${status}`]
```

Comment:
Updated this logic so that we don't attempt to parse JSON when the status is FAIL (this will lead to a parsing error)